### DISCUSSION TITLE
Coloring different subdomains in different colors

### DISCUSSION THREAD

**Discussion by famrein** [BradyAJohnston/MolecularNodes#269] [Q&A] [answered]
I have a protein which has 3 subdomains, with different residue ID ranges. My goal is to color every domain differently, ideally in a cartoon representation. I think this might be easier with spheres but anyways. Disclaimer: im very new at blender. Right now I have this sort of thing in the geometry nodes, where I'm handling each domain separately and then join them at the end. I have attached a picture, only 2 domains are pictured.  The problem with this is that I don't get a smooth joining between domains (see second attached image). Is there a) a better, cleaner way to colour 3 subdomains, b) not a better way, but a way to make a smoother joining of the domains?

Thanks!

![image](https://github.com/BradyAJohnston/MolecularNodes/assets/22937598/3a9d8b49-c4ce-42f1-a4d2-b5e28c6338a2)
![image](https://github.com/BradyAJohnston/MolecularNodes/assets/22937598/62b2762f-a48c-4211-ad5d-5570a23701ce)

**Comment by BradyAJohnston** [comment, marked as answer]
Hi @famrein thanks for the question! Currently there is some smoothing abblied to the cartoon, so if you 'break the chain' by having different cartoon representations applied to different selections, then it might not line up like you are seeing. 

What I would recommend is to have a single cartoon style applied, but instead selectively apply 3 different colours to the atoms before you create the cartoon style.

I have an example in the screenshot below where I use the `MOL_color_set` node to set the color of different selections, then apply the style. You can then see in the screenshot that they are part of the same mesh, and the colour blends between the different domains.

Hope this helps! Let me know if you have more questions.

![image](https://github.com/BradyAJohnston/MolecularNodes/assets/36021261/3929e8b6-4c37-4009-93b0-1affb0968710)

![image](https://github.com/BradyAJohnston/MolecularNodes/assets/36021261/ec026e81-91bc-453d-a0f9-c4ca9d273612)

**Reply by famrein** [reply]
Thats exactly it, perfect! Thanks

**Reply by BradyAJohnston** [reply]
Glad it works :)
